Close/CLOSE #17390 Products API, GET /products without variants

kastoras opened the pull request:
When accessing get /products, using $without_variants parameter to true (the deafult value is false), the api return only products and no variants.

# Close #17390
When calling Rest API get products, I am getting all products & variants

# New, Get only Products without variants
I added one new parameter $without_variants when calling GET /products. When this parameter is true I added a new SQL statment ($sql .= ' AND t.rowid in (SELECT distinct fk_product_parent FROM '.MAIN_DB_PREFIX.'product_attribute_combination')
Using this sql condition, the inner select query will gather all product IDs that act as a parrent product for variant products.
Finaly using t.rowid IN in the above IDs the api returns only products and not variants.

kastoras committed:
Close/CLOSE #17390 Products API, GET /products without variants

When accessing get /products, using $without_variants parameter to true (the deault value is false), the api retur only products and no variants.

stickler-ci committed:
Fixing style errors.

eldy committed:
Update api_products.class.php

eldy commented on htdocs/product/class/api_products.class.php:
```diff
@@ -186,6 +187,11 @@ public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $p
 		$sql .= ' WHERE t.entity IN ('.getEntity('product').')';
+
+		if ($without_variants) {
+			$sql .= ' AND t.rowid in (select distinct fk_product_parent from '.MAIN_DB_PREFIX.'product_attribute_combination)';
```
Some products are A) common products, some are B) parent of variant and some are C) variant.
Here adding the test will return all products B (right).
Why not but it means label of parameter is wrong (without_variant means A for me).
If what you want if really B, i suggest to rename parameter into "$parent_of_variant"

Or better, rename it into "$variant_filter" with 2 values (0 = all, 1=A only, 2=B only, 3=C only). Need more change. First suggestion correction will be enough.

kastoras replied:
Good point. I will make the change and push it again.

kastoras committed:
Close/CLOSE #17390 Products API, GET /products variants filter

Added new parameter for variants filter. Using this filter you can get products without variants only, parrent products only, child products only.

kastoras commented:
I have implemented this using sub-queries. I could do this using joins if this is more preferable.
